Doc/library/sqlite3.rst

```diff
@@ -2358,6 +2330,84 @@ can be found in the `SQLite URI documentation`_.
 .. _SQLite URI documentation: https://www.sqlite.org/uri.html
 
 
+.. _sqlite3-row-factory-how-to:
+
+How to work with row factories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+By default, :mod:`!sqlite3` represent fetched rows as :class:`tuples <tuple>`.
+If a :class:`!tuple` does not suit your needs,
+use the built-in :class:`Row` type or a custom :attr:`~Connection.row_factory`.
+The former provides both indexed and case-insensitive named access to columns,
+with low memory overhead and minimal performance impact.
+Example use:
+
+.. doctest::
+
+   >>> con = sqlite3.connect(":memory:")
+   >>> con.row_factory = sqlite3.Row
+   >>> res = con.execute("SELECT 'Earth' AS name, 6378 AS radius")
+   >>> row = res.fetchone()
+   >>> row.keys()
+   ['name', 'radius']
+   >>> row[0], row["name"]  # Access by index and name.
```

Comment:
index or name ?

Reply:
Access by index and access by name.

Reply:
Sorry, I meant: say `Access by index or name`, not `and`

Reply:
Hm, I think that depends on how you view it 😉 The first element of the tuple comes from indexed access and the second element comes from access by name.

Reply:
However, I'm not sure it matters much. I'm sure the message gets through no matter the conjunction; I'm fine with either.

Reply:
I find @merwok 's version clearer; access by index _and_ name could be read to imply that both can or must be used together, whereas it is an (exclusive) or—only one can be used at a time. Using "or" avoids this potential ambiguity, and is AFAIK more idiomatic for this usage.

As an actual suggestion:

```suggestion
   >>> row[0], row["name"]  # Access by index or name.
```

Also, though it was slightly confusing to me that these two were in a tuple on first reading rather than as separate lines with separate outputs (as they would be used in reality), which would make the above clearer, IMO.

Reply:
> Also, though it was slightly confusing to me that these two were in a tuple on first reading rather than as separate lines with separate outputs (as they would be used in reality), which would make the above clearer, IMO.

Yes, I thought about that yesterday; it is a weird example using a tuple like this. I think I'll resolve it by putting them on separate lines.